components/AccountBalance.js

```diff
@@ -2,37 +2,90 @@ import { useEffect, useState } from "react";
 import { formatNumber } from "@/lib/utils";
 import styled from "styled-components";
 
-export default function AccountBalance({ transactions }) {
+export default function AccountBalance({
+  transactions,
+  filteredAccount,
+  filteredAccountType,
+}) {
   const [currentBalance, setCurrentBalance] = useState(0);
 
   useEffect(() => {
-    const sumOfTransaction = transactions.reduce((total, transaction) => {
-      return transaction.type === "income"
-        ? total + transaction.amount
-        : total - transaction.amount;
-    }, 0);
-    setCurrentBalance(sumOfTransaction);
+    const incomeTotal = transactions
+      .filter((transaction) => transaction.type === "income")
+      .reduce((total, transaction) => total + transaction.amount, 0);
+
+    const expenseTotal = transactions
+      .filter((transaction) => transaction.type === "expense")
+      .reduce((total, transaction) => total + transaction.amount, 0);
+
+    setCurrentBalance(incomeTotal - expenseTotal);
   }, [transactions]);
 
   const currentBalanceType = currentBalance < 0 ? "expense" : "income";
 
+  function getTitle() {
+    if (filteredAccountType === "income") return "Total Income";
+    if (filteredAccountType === "expense") return "Total Expense";
+    if (filteredAccountType === "total") return "Total sum";
+    return "Current Balance";
+  }
```

Comment:
nice solution! there is one that is a bit more concise using an object: 
```js
const titleMap = {
  income: "Total Income",
  expense: "Total Expense",
  total: "Total Sum"
}

const title = titleMap[filteredAccountType] || "Current Balance"
```
Feel free to keep your solution or switch to this one.

Reply:
I really like this solution! I have taken this solution.